src/isar/apis/schedule/scheduling_controller.py

```diff
@@ -98,6 +103,23 @@ def start_mission(
         self.scheduling_utilities.start_mission(mission=mission)
         return self._api_response(mission)
 
+    def return_home(self) -> StartMissionResponse:
+        self.logger.info("Received request to return home")
+
+        state: States = self.scheduling_utilities.get_state()
+        self.scheduling_utilities.verify_state_machine_ready_to_receive_return_home_mission(
+            state
+        )
+
+        mission = Mission(
+            tasks=[ReturnToHome()],
+            name="Return Home",
+        )
+
+        self.logger.info(f"Starting mission: {mission.id}")
+        self.scheduling_utilities.start_mission(mission=mission)
+        return self._api_response(mission)
```

Comment:
I see that you are sending return home as a normal mission and then checking it in the state machine what kind of mission you get and reacting accordingly. Wouldn't it be cleaner to instead have this be a different event sent to the state machine which specifically tells it to start a return home mission? That way we can have a specific event handler for the event in robotStandingStill for example, instead of reusing other events.

Reply:
Agreed, introduced a return_home event that address this. The checking of whether it is a return home mission is kept as it is currently used so that stopping can differentiate what kind of mission have been stopped